botorch/acquisition/preference.py

```diff
@@ -119,6 +131,76 @@ def forward(self, X: Tensor) -> Tensor:
         return acqf_val
 
 
+class qExpectedUtilityOfBestOption(MCAcquisitionFunction):
+    r"""MC-based Expected Utility of Best Option (qEUBO)
+
+    This computes qEUBO by
+    (1) sampling the joint posterior over q points
+    (2) evaluating the maximum objective value accross the q points
+    (3) averaging over the samples
+
+    `qEUBO(X) = E[max Y], Y ~ f(X), where X = (x_1,...,x_q)`
+    """
+
+    def __init__(
+        self,
+        pref_model: Model,
+        outcome_model: Optional[DeterministicModel] = None,
+        sampler: Optional[MCSampler] = None,
+        objective: Optional[MCAcquisitionObjective] = None,
+        posterior_transform: Optional[PosteriorTransform] = None,
+        X_pending: Optional[Tensor] = None,
+    ) -> None:
+        r"""MC-based Expected Utility of Best Option (qEUBO) as proposed
+        in [Astudillo2023qeubo]_.
+
+        Args:
+            pref_model: The preference model that maps the outcomes (i.e., Y) to
+                scalar-valued utility.
+            outcome_model: A deterministic model that maps parameters (i.e., X) to
+                outcomes (i.e., Y). The outcome model f defines the search space of
+                Y = f(X). If model is None, we are directly calculating qEUBO on
+                the parameter space.
+             sampler: The sampler used to draw base samples. See `MCAcquisitionFunction`
+                more details.
+            objective: The MCAcquisitionObjective under which the samples are evaluated.
+                Defaults to `IdentityMCObjective()`.
+            posterior_transform: A PosteriorTransform (optional).
+            X_pending:  A `m x d`-dim Tensor of `m` design points that have been
+                submitted for function evaluation but have not yet been evaluated.
+                Concatenated into X upon forward call. Copied and set to have no gradient.
```

Comment:
```suggestion
                Concatenated into X upon forward call. Copied and set
                to have no gradient.
```
Line length lint here